x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -173,6 +171,36 @@ public RollupIndexerAction.ShardRollupResponse execute() throws IOException {
         return new RollupIndexerAction.ShardRollupResponse(indexShard.shardId(), numIndexed.get());
     }
 
+    /**
+     * Retrieve field value fetchers for a list of fields.
+     */
+    private static List<FieldValueFetcher> getFieldValueFetchers(SearchExecutionContext context, String[] fields) {
+        List<FieldValueFetcher> fetchers = new ArrayList<>();
+        for (String field : fields) {
+            MappedFieldType fieldType = context.getFieldType(field);
+            assert fieldType != null : "Unknown field type for field: [" + field + "]";
+            fetchers.add(fieldType.fieldValueFetcher(context));
+        }
+        return Collections.unmodifiableList(fetchers);
+    }
+
+    private static Map<String, FormattedDocValues> docValuesFetchers(
+        LeafReaderContext ctx,
+        SearchExecutionContext searchContext,
+        List<FieldValueFetcher> fieldValueFetchers
+    ) {
+        final Map<String, FormattedDocValues> docValuesFetchers = new LinkedHashMap<>(fieldValueFetchers.size());
+        for (FieldValueFetcher fetcher : fieldValueFetchers) {
+            for (Map.Entry<String, FormattedDocValues> e : fetcher.getLeaves(ctx).entrySet()) {
+                if (searchContext.fieldExistsInIndex(e.getKey())) {
```

Comment:
If field does not exist in the index, it will be ignored even if it exists in the mapping. This should improve performance for use cases such as ECS